🐛 Bug Report: Creating customs test, can't import @backstage/plugin-scaffolder-node-test-utils

### 📜 Description

I want to create a test for my custom action. After reading the [docs](https://backstage.io/docs/features/software-templates/writing-tests-for-actions) on how to write tests, i ran into an issue.

When creating the test the `createMockActionContext` from `@backstage/plugin-scaffolder-node-test-utils` is not found. I can find the package in the node_modules, but the `createMockActionContext` can't be used.

The error is as follows `Cannot find module '@backstage/plugin-scaffolder-node-test-utils' from 'plugins/scaffolder/actions/merge-azure-pipeline.test.ts'` i've listed the reproduction steps below.

### 👍 Expected behavior

It should find the module `@backstage/plugin-scaffolder-node-test-utils` and i should be able to use the `createMockActionContext`

### 👎 Actual Behavior with Screenshots

Cannot find module '@backstage/plugin-scaffolder-node-test-utils' from 'plugins/scaffolder/actions/my-test.test.ts

### 👟 Reproduction steps

`npx @backstage/create-app@latest`

cd into the project

from the root of the project run:
`yarn --cwd packages/backend add @backstage/backend-test-utils @backstage/plugin-scaffolder-node-test-utils`

create the file `packages/backend/src/plugins/scaffolder/actions/my-test.test.ts`

add the following content to the file
```ts
import { createMockDirectory } from '@backstage/backend-test-utils';
import { createMockActionContext } from '@backstage/plugin-scaffolder-node-test-utils';
import { createMyAction } from './my-action';

describe('test:action', () => {
  const workspacePath = createMockDirectory().resolve('workspace');
  const mockContext = createMockActionContext({
    workspacePath,
  });
  it('should foo', async () => {
    const action = createMyAction();
    await action.handler({
      ...mockContext,
      input: {
        myInput: 'foo',
      },
    });

    expect(true).toBe(true);
  });
});
```

When running `yarn test my-test` i'm getting the following error:

`Cannot find module '@backstage/plugin-scaffolder-node-test-utils' from 'plugins/scaffolder/actions/my-test.test.ts'`

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

OS:   Linux 5.15.133.1-microsoft-standard-WSL2 - linux/x64
node: v20.10.0
yarn: 1.22.21
cli:  0.25.2 (installed)
backstage:  1.23.0

Dependencies:
  @backstage/app-defaults                                          1.5.0
  @backstage/backend-app-api                                       0.5.14
  @backstage/backend-common                                        0.21.3
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.6
  @backstage/backend-plugin-api                                    0.6.13
  @backstage/backend-tasks                                         0.5.18
  @backstage/backend-test-utils                                    0.3.3
  @backstage/catalog-client                                        1.6.0
  @backstage/catalog-model                                         1.4.4, 1.4.5-next.0
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.3
  @backstage/cli                                                   0.25.2
  @backstage/config-loader                                         1.6.2
  @backstage/config                                                1.1.1, 1.1.2-next.0
  @backstage/core-app-api                                          1.12.0
  @backstage/core-compat-api                                       0.2.0
  @backstage/core-components                                       0.14.0
  @backstage/core-plugin-api                                       1.9.0
  @backstage/e2e-test-utils                                        0.1.1
  @backstage/errors                                                1.2.3, 1.2.4-next.0
  @backstage/eslint-plugin                                         0.1.5
  @backstage/frontend-plugin-api                                   0.6.0
  @backstage/integration-aws-node                                  0.1.9
  @backstage/integration-react                                     1.1.24
  @backstage/integration                                           1.9.0
  @backstage/plugin-api-docs                                       0.11.0
  @backstage/plugin-app-backend                                    0.3.61
  @backstage/plugin-app-node                                       0.1.13
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.5
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.4
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.8
  @backstage/plugin-auth-backend-module-github-provider            0.1.10
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.10
  @backstage/plugin-auth-backend-module-google-provider            0.1.10
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.8
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.10
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.6
  @backstage/plugin-auth-backend-module-oidc-provider              0.1.3
  @backstage/plugin-auth-backend-module-okta-provider              0.0.6
  @backstage/plugin-auth-backend                                   0.21.3
  @backstage/plugin-auth-node                                      0.4.8
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.10
  @backstage/plugin-catalog-backend                                1.17.3
  @backstage/plugin-catalog-common                                 1.0.21
  @backstage/plugin-catalog-graph                                  0.4.0
  @backstage/plugin-catalog-import                                 0.10.6
  @backstage/plugin-catalog-node                                   1.7.3
  @backstage/plugin-catalog-react                                  1.10.0
  @backstage/plugin-catalog                                        1.17.0
  @backstage/plugin-events-node                                    0.2.22
  @backstage/plugin-github-actions                                 0.6.11
  @backstage/plugin-org                                            0.6.20
  @backstage/plugin-permission-common                              0.7.12, 0.7.13-next.0
  @backstage/plugin-permission-node                                0.7.24
  @backstage/plugin-permission-react                               0.4.20
  @backstage/plugin-proxy-backend                                  0.4.11
  @backstage/plugin-scaffolder-backend-module-azure                0.1.5
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.3
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.3
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.3
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.5
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.3
  @backstage/plugin-scaffolder-backend-module-github               0.2.3
  @backstage/plugin-scaffolder-backend-module-gitlab               0.2.16
  @backstage/plugin-scaffolder-backend                             1.21.3
  @backstage/plugin-scaffolder-common                              1.5.0, 1.5.1-next.0
  @backstage/plugin-scaffolder-node-test-utils                     0.1.0-next.0
  @backstage/plugin-scaffolder-node                                0.3.3
  @backstage/plugin-scaffolder-react                               1.8.0
  @backstage/plugin-scaffolder                                     1.18.0
  @backstage/plugin-search-backend-module-catalog                  0.1.17
  @backstage/plugin-search-backend-module-pg                       0.5.22
  @backstage/plugin-search-backend-module-techdocs                 0.1.17
  @backstage/plugin-search-backend-node                            1.2.17
  @backstage/plugin-search-backend                                 1.5.3
  @backstage/plugin-search-common                                  1.2.10
  @backstage/plugin-search-react                                   1.7.6
  @backstage/plugin-search                                         1.4.6
  @backstage/plugin-tech-radar                                     0.6.13
  @backstage/plugin-techdocs-backend                               1.9.6
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.5
  @backstage/plugin-techdocs-node                                  1.11.5
  @backstage/plugin-techdocs-react                                 1.1.16
  @backstage/plugin-techdocs                                       1.10.0
  @backstage/plugin-user-settings                                  0.8.1
  @backstage/release-manifests                                     0.0.11
  @backstage/repo-tools                                            0.6.3
  @backstage/test-utils                                            1.5.0
  @backstage/theme                                                 0.5.1
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

I encountered the same issue and already spent a few hours on it. 

package.json
```
  "devDependencies": {
    "@backstage/backend-test-utils": "^0.3.3",
    "@backstage/cli": "^0.25.0",
    "@backstage/plugin-scaffolder-node-test-utils": "^0.1.0-next.0",
    "@types/dockerode": "^3.3.0",
    "@types/express": "^4.17.6",
    "@types/express-serve-static-core": "^4.17.5",
    "@types/luxon": "^2.0.4"
  },
```

```
Cannot find module '@backstage/plugin-scaffolder-node-test-utils' from 'plugins/scaffolder/actions/folder-copy.test.ts'
```

--- a/.changeset/slimy-bananas-brush.md
+++ b/.changeset/slimy-bananas-brush.md
@@ -2,4 +2,4 @@
 '@backstage/plugin-scaffolder-node-test-utils': patch
 ---
 
-Updated publishConfig.main entry in package.json from `dist/index.esm.js` to `dist/index.cjs.js` for CommonJS compatibility.
+Fix issue with package bundling, should be `dist/index.cjs.js` instead of `dist/index.esm.js`.

--- a/.changeset/slimy-bananas-brush.md
+++ b/.changeset/slimy-bananas-brush.md
@@ -1,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-node-test-utils': patch
+---
+
+Updated publishConfig.main entry in package.json from `dist/index.esm.js` to `dist/index.cjs.js` for CommonJS compatibility.

--- a/plugins/scaffolder-node-test-utils/package.json
+++ b/plugins/scaffolder-node-test-utils/package.json
@@ -6,7 +6,7 @@
   },
   "publishConfig": {
     "access": "public",
-    "main": "dist/index.esm.js",
+    "main": "dist/index.cjs.js",
     "types": "dist/index.d.ts"
   },
   "repository": {

